tests/integration/security/util/authn/authn_util.go

```diff
@@ -52,30 +52,29 @@ func (c *TestCase) String() string {
 
 // CheckAuthn checks a request based on ExpectResponseCode.
 func (c *TestCase) CheckAuthn(responses echoclient.Responses, err error) error {
-	if len(responses) == 0 {
-		return fmt.Errorf("%s: no response", c)
-	}
-	if responses[0].Code != c.ExpectResponseCode {
-		return fmt.Errorf("%s: got response code %s, err %v", c, responses[0].Code, err)
-	}
-	// Checking if echo backend see header with the given value by finding them in response body
-	// (given the current behavior of echo convert all headers into key=value in the response body)
-	for k, v := range c.ExpectHeaders {
-		matcher := fmt.Sprintf("%s=%s", k, v)
-		if len(v) == 0 {
-			if strings.Contains(responses[0].Body, matcher) {
-				return fmt.Errorf("%s: expect header %s does not exist, got response\n%s", c, k, responses[0].Body)
+	return check.And(
+		check.StatusCode(c.ExpectResponseCode),
+		check.Each(func(r echoclient.Response) error {
+			// Checking if echo backend see header with the given value by finding them in response body
+			// (given the current behavior of echo convert all headers into key=value in the response body)
+			for k, v := range c.ExpectHeaders {
```

Comment:
outside scope but this should use Key() probably

Reply:
Yeah, I'm going to address this in https://github.com/istio/istio/pull/37358